src/source-pcap-file-helper.c

```diff
@@ -160,6 +168,23 @@ TmEcode PcapFileDispatch(PcapFileFileVars *ptv)
     SCReturnInt(loop_result);
 }
 
+/** \internal
+ *  \brief get the timestamp of the first packet and rewind
+ *  \retval bool true on success, false on error
+ */
+static bool PeekFirstPacketTimestamp(PcapFileFileVars *pfv)
+{
+    int r = pcap_next_ex(pfv->pcap_handle, &pfv->first_pkt_hdr, &pfv->first_pkt_data);
+    if (r <= 0 || pfv->first_pkt_hdr == NULL) {
+        SCLogError(SC_ERR_PCAP_OPEN_OFFLINE,
+                "failed to get first packet timestamp. pcap_next_ex(): %d", r);
+        return false;
+    }
+    pfv->first_pkt_ts.tv_sec = pfv->first_pkt_hdr->ts.tv_sec;
```

Comment:
Suggest  `pfv->first_pkt_ts = pfv->first_pkt_hdr->ts`

Reply:
These are not guaranteed to be the same. More specifically, on OpenBSD they are not.